### DISCUSSION TITLE
Imagemagick crop and rename throug subfolder

### DISCUSSION THREAD

**Discussion by Cenglert174** [ImageMagick/ImageMagick#2264] [Help]
Hi,
i like reading about scrips, but i am too stupid to write or understand them.
There are many folders with scanned Book-Pages and i want them to be croped and renamed.
I found

 `@echo off
pushd "%~dp1"

convert "%~1" ^
   -crop 2x1@ -write mpr:img -delete 0--1 ^
   mpr:img[0] -write "%~n1-L.tif" +delete ^
   mpr:img[1] "%~n1-R.tif"

popd
exit /b`

Maybe from here.
This works fine with one Picture by Picture.

What can i do to run this over many subfolders?
System is Win 10 with ImageMagick-7.0.10-Q16-HDRI registered in system path.
Any Idea?

**Comment by snibgo** [comment]
Your question is about Windows BAT scripting, not specifically ImageMagick.

You could call that script inside another command or script that runs a "for" loop. For example, assuming your script is called "splitbook.bat", you could have an interactive command like this:
```
for /F "usebackq" %L in (`dir /s /b *.png`) do call splitbook.bat %L
```
But please note that questions about scripting should be addressed to a more suitable forum.

**Comment by Cenglert174** [comment]
Thanks for your answer,
when i try to run this, there is a problem with "L".
I also changed .png in .tif
I tryed "%L" and %%L and "%%L" with now result but i dont know anything about these signs.
It is all try and error.

Do you know a mor suitable forum for my question?
